### PR TITLE
KAFKA-12321 the comparison function for uuid type should be 'equals' …

### PR DESCRIPTION
**Pull request by chia7712**
issue: https://issues.apache.org/jira/browse/KAFKA-12321

### current comparison
```java
if (this.taggedUuid != Uuid.fromString("H3KKO4NTRPaCWtEmm3vW7A"))
```

### fixed comparison
```java
if (!this.taggedUuid.equals(Uuid.fromString("H3KKO4NTRPaCWtEmm3vW7A")))
```

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/test/java/org/apache/kafka/common/message/MessageTest.java [hunk 1]
```diff
@@ -749,6 +750,13 @@ public void testProduceResponseVersions() throws Exception {
         }
     }
 
+    @Test
+    public void defaultValueShouldBeWritable() {
+        for (short version = SimpleExampleMessageData.LOWEST_SUPPORTED_VERSION; version <= SimpleExampleMessageData.HIGHEST_SUPPORTED_VERSION; ++version) {
+            MessageUtil.toByteBuffer(new SimpleExampleMessageData(), version);
```
Comment (ijuma) [inline clients/src/test/java/org/apache/kafka/common/message/MessageTest.java:756]: This fails without the change in this PR?

**Review by ijuma — APPROVED** [review]
Good catch, LGTM assuming the question below is true. We should also backport it to older branches.

File: clients/src/test/java/org/apache/kafka/common/message/MessageTest.java (hunk 1 above)
Comment (chia7712) [reply clients/src/test/java/org/apache/kafka/common/message/MessageTest.java:756]: > This fails without the change in this PR?

yep. The generated code is shown below.
```java
            if (this.taggedUuid != Uuid.fromString("H3KKO4NTRPaCWtEmm3vW7A")) {
                throw new UnsupportedVersionException("Attempted to write a non-default taggedUuid at version " + _version);
            }
```

**Comment by chia7712** [comment]
push to trunk, 2.8, 2.7 and 2.6

**Pull request merged**